pkg/storaged/dialog.jsx

```diff
@@ -724,100 +725,50 @@ export const SelectSpaces = (tag, title, options) => {
+                            const selected = (val.indexOf(spc) >= 0);
+                            const block = spc.block ? nice_block_name(spc.block) : "";
+
+                            const on_change = (_event, checked) => {
+                                if (checked && !selected)
+                                    change(val.concat(spc));
+                                else if (!checked && selected)
+                                    change(val.filter(v => (v != spc)));
+                            };
 
-        render: (val, change) => {
-            if (options.spaces.length === 0)
-                return <span className="text-danger">{options.empty_warning}</span>;
-
-            return (
-                <DataList isCompact
-                    data-field={tag} data-field-type="select-spaces">
-                    { options.spaces.map(spc => {
-                        const block = spc.block ? nice_block_name(spc.block) : "";
-                        const desc = block === spc.desc ? "" : spc.desc;
-                        const on_change = (event) => {
-                            if (event.target.checked)
-                                change(spc);
-                        };
-
-                        return (
-                            <DataListItem key={spc.block ? spc.block.Device : spc.desc}>
-                                <DataListItemRow>
-                                    <div className="pf-v5-c-data-list__item-control">
-                                        <div className="pf-v5-c-data-list__check">
-                                            <input type='radio' value={desc} name='space' checked={val == spc} onChange={on_change} />
-                                        </div>
-                                    </div>
-                                    <DataListItemCells
-                                        dataListCells={[
-                                            <DataListCell key="select-space-name" className="select-space-name">
-                                                {format_size_and_text(spc.size, desc)}
-                                            </DataListCell>,
-                                            <DataListCell alignRight isFilled={false} key="select-space-details" className="select-space-details">
-                                                {block}
-                                            </DataListCell>,
-                                        ]}
-                                    />
-                                </DataListItemRow>
-                            </DataListItem>
-                        );
-                    })
-                    }
-                </DataList>
+                            return (
+                                <Tr key={spc.block ? spc.block.Device : spc.desc}
+                                    onRowClick={() => on_change(null, !selected)}>
+                                    <Td select={{
```

Comment:
These 4 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18967-20230629-132526-4d4c19dd-fedora-38-devel-expensive//Coverage/lcov/github-pr.diff.gcov.html)